packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -335,6 +334,14 @@ class _MonthPickerState extends ScrollableWidgetListState<MonthPicker> {
     return result;
   }
 
+  Widget build(BuildContext context) {
+    return new ScrollableLazyList(
+      itemExtent: config.itemExtent,
+      itemCount: (config.lastDate.year - config.firstDate.year) * 12 + config.lastDate.month - config.firstDate.month + 1,
```

Comment:
Yeah, we can support infinite ranges.  We want finite ranges too, though, so people can make constrained pickers.